python/python/glide/async_commands/redis_modules/search/field.py

```diff
@@ -0,0 +1,385 @@
+from enum import Enum
+from typing import Dict, List, Optional
+
+
+class SortableOption(Enum):
+    """
+    Options for fields with SORTABLE argument.
+    """
+
+    SORTABLE = "SORTABLE"
+    """
+    Sets the field as sortable.
+    """
+    SORTABLE_UNF = "SORTABLE UNF"
+    """
+    Sets the field as sortable, disables the normalization and keep the original form of the field value.
+    """
+
+
+class PhoneticType(Enum):
+    """
+    Activate phonetic matching for text searches, specifies the phonetic algorithm and language used.
+    """
+
+    DM_ENGLISH = "dm:en"
+    """
+    Double metaphone for English.
+    """
+    DM_FRENCH = "dm:fn"
+    """
+    Double metaphone for French.
+    """
+    DM_PORTUGUESE = "dm:pt"
+    """
+    Double metaphone for Portuguese
+    """
+    DM_SPANISH = "dm:es"
+    """
+    Double metaphone for Spanish.
+    """
+
+
+class CoordinateSystem(Enum):
+    """
+    Specifies the coordinate system used for geo fields.
+    """
+
+    SPHERICAL = "SPHERICAL"
+    """
+    Spherical coordinates system used with latitude/longitude.
+    """
+    FLAT = "FLAT"
+    """
+    Flat coordinates system used with XY coordinates.
+    """
+
+
+class VectorAlgorithm(Enum):
+    """
+    Specifies the algorithm used for vector similarity search.
+    """
+
+    FLAT = "FLAT"
+    """
+    Brute-force index. Used for flat vector indexing.
+    """
+    HNSW = "HNSW"
+    """
+    Modified version of nmslib/hnswlib, which is an implementation of Efficient and robust 
+    approximate nearest neighbor search using Hierarchical Navigable Small World graphs.
+    """
+
+
+class Field:
```

Comment:
If I understand it right the Field class shouldn't be used with create_index, only the extended versions of it.
If so, please make it an abstract class, so the get_field_args would be an abstract function and instead have another function implemented only in the base class `get_filed_identifier` which will return the filed name and alias